Arch Linux does not build under Kernel 6.12.1

Hello,  
using the AUR package the module installs fine for the lts kernel 6.6.63, but throws this error on the standard kernel:
```
==> dkms install --no-depmod universal-pidff/0.0.10.r0.gd95b73e -k 6.12.1-arch1-1

Error! Bad return status for module build on kernel: 6.12.1-arch1-1 (x86_64)
Consult /var/lib/dkms/universal-pidff/0.0.10.r0.gd95b73e/build/make.log for more information.
==> WARNING: `dkms install --no-depmod universal-pidff/0.0.10.r0.gd95b73e -k 6.12.1-arch1-1' exited 10
```

The log file contains this:
```
DKMS make.log for universal-pidff-0.0.10.r0.gd95b73e for kernel 6.12.1-arch1-1 (x86_64)
Wed Nov 27 02:22:06 CET 2024
make -C /lib/modules/6.12.1-arch1-1/build M=$PWD
make[1]: Entering directory '/usr/lib/modules/6.12.1-arch1-1/build'
  CC [M]  /var/lib/dkms/universal-pidff/0.0.10.r0.gd95b73e/build/hid-pidff-wrapper.o
  CC [M]  /var/lib/dkms/universal-pidff/0.0.10.r0.gd95b73e/build/hid-pidff.o
/var/lib/dkms/universal-pidff/0.0.10.r0.gd95b73e/build/hid-pidff-wrapper.c:166:25: error: initialization of 'const __u8 * (*)(struct hid_device *, __u8 *, unsigned int *)' {aka 'const unsigned char * (*)(struct hid_device *, unsigned char *, unsigned int *)'} from incompatible pointer type 'u8 * (*)(struct hid_device *, __u8 *, unsigned int *)' {aka 'unsigned char * (*)(struct hid_device *, unsigned char *, unsigned int *)'} [-Wincompatible-pointer-types]
  166 |         .report_fixup = universal_pidff_report_fixup
      |                         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
/var/lib/dkms/universal-pidff/0.0.10.r0.gd95b73e/build/hid-pidff-wrapper.c:166:25: note: (near initialization for 'universal_pidff.report_fixup')
make[3]: *** [scripts/Makefile.build:229: /var/lib/dkms/universal-pidff/0.0.10.r0.gd95b73e/build/hid-pidff-wrapper.o] Error 1
make[3]: *** Waiting for unfinished jobs....
make[2]: *** [/usr/lib/modules/6.12.1-arch1-1/build/Makefile:1936: /var/lib/dkms/universal-pidff/0.0.10.r0.gd95b73e/build] Error 2
make[1]: *** [Makefile:224: __sub-make] Error 2
make[1]: Leaving directory '/usr/lib/modules/6.12.1-arch1-1/build'
make: *** [Makefile:5: default] Error 2
```

`dkms status` only reports the lts kernel, and that as successfully installed.  
  
Although, while trying to troubleshoot this, `dkms status` returned also this row:
`universal-pidff/0.0.6.r4.g9e27df1: broken` (while the lts kernel uses version `0.0.10.r0.gd95b73e`)  
So, as a troubleshooting step I uninstalled the aur package, removed the broken version, which leaves us at the current state.
  
Other Troubleshooting step like reinstalling and rebooting did not resolve the issue either.

It looks like I have the same problem. I'm trying to build it on Fedora 41.

`make -C /lib/modules/`uname -r`/build M=$PWD
make[1]: вход в каталог «/usr/src/kernels/6.11.10-300.fc41.x86_64»
  CC [M]  /opt/universal-pidff/hid-pidff-wrapper.o
/opt/universal-pidff/hid-pidff-wrapper.c:166:25: ошибка: initialization of «__u8 * (*)(struct hid_device *, __u8 *, unsigned int *)» {aka «unsigned char * (*)(struct hid_device *, unsigned char *, unsigned int *)»} from incompatible pointer type «const u8 * (*)(struct hid_device *, __u8 *, unsigned int *)» {aka «const unsigned char * (*)(struct hid_device *, unsigned char *, unsigned int *)»} [-Wincompatible-pointer-types]
  166 |         .report_fixup = universal_pidff_report_fixup
      |                         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
/opt/universal-pidff/hid-pidff-wrapper.c:166:25: замечание: (где-то рядом с инициализацией для «universal_pidff.report_fixup»)
make[3]: *** [scripts/Makefile.build:244: /opt/universal-pidff/hid-pidff-wrapper.o] Ошибка 1
make[2]: *** [/usr/src/kernels/6.11.10-300.fc41.x86_64/Makefile:1966: /opt/universal-pidff] Ошибка 2
make[1]: *** [Makefile:236: __sub-make] Ошибка 2
make[1]: выход из каталога «/usr/src/kernels/6.11.10-300.fc41.x86_64»
make: *** [Makefile:5: default] Ошибка 2
`

--- a/dkms.conf
+++ b/dkms.conf
@@ -1,5 +1,5 @@
 PACKAGE_NAME="universal-pidff"
-PACKAGE_VERSION="0.0.11"
+PACKAGE_VERSION="0.0.12"
 MAKE[0]="make KVERSION=$kernelver"
 CLEAN="make clean"
 BUILT_MODULE_NAME[0]="hid-universal-pidff"

--- a/hid-pidff-wrapper.c
+++ b/hid-pidff-wrapper.c
@@ -11,6 +11,7 @@
 #include <linux/hid.h>
 #include <linux/module.h>
 #include <linux/input-event-codes.h>
+#include <linux/version.h>
 #include "hid-ids.h"
 #include "hid-pidff.h"
 
@@ -52,24 +53,32 @@ static const struct hid_device_id pidff_wheel_devices[] = {
 MODULE_DEVICE_TABLE(hid, pidff_wheel_devices);
 
 
-static const u8 *moza_report_fixup(struct hid_device *hdev, __u8 *rdesc,
-				   unsigned int *rsize)
+static
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(6,12,0)
+const
+#endif
+u8 *moza_report_fixup(struct hid_device *hdev, __u8 *rdesc,
+		      unsigned int *rsize)
 {
 	// Fix data type on PID Device Control
-        if (rdesc[1002] == 0x91 && rdesc[1003] == 0x02) {
+	if (rdesc[1002] == 0x91 && rdesc[1003] == 0x02) {
 		rdesc[1003] = 0x00; // Fix header, it needs to be Array.
-        }
-        return rdesc;
+	}
+	return rdesc;
 }
 
 
-static const u8 *universal_pidff_report_fixup(struct hid_device *hdev, __u8 *rdesc,
-        				      unsigned int *rsize)
+static
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(6,12,0)
+const
+#endif
+u8 *universal_pidff_report_fixup(struct hid_device *hdev, __u8 *rdesc,
+        			 unsigned int *rsize)
 {
-		if (hdev->vendor == USB_VENDOR_ID_MOZA) {
-			return moza_report_fixup(hdev, rdesc, rsize);
-		}
-		return rdesc;
+	if (hdev->vendor == USB_VENDOR_ID_MOZA) {
+		return moza_report_fixup(hdev, rdesc, rsize);
+	}
+	return rdesc;
 }
 
 /*

